File: Model/Source/WaterAnalysis.vb
Imports System.ComponentModel
Imports Wells.Model.ReflectionExtension

Public Class WaterAnalysis
    Inherits ChemicalAnalysis

    Sub New()
        MyBase.New()
        SampleOf = SampleType.Water
    End Sub

    Sub New(well As Well)
        MyBase.New(well)
        SampleOf = SampleType.Water
    End Sub

    <DisplayName("Pozo"), Browsable(True)>
    Overrides Property WellName As String
        Get
            If Well IsNot Nothing Then
                Return Well.Name
            Else
                Return _WellName
            End If
        End Get
        Set
            _WellName = Value
        End Set
    End Property

    <DisplayName("Fecha"), Browsable(True)>
    Overrides ReadOnly Property RealDate As Date
        Get
            Return Date.ParseExact(SampleDate, "dd/MM/yyyy", Nothing)
        End Get
    End Property

    <DisplayName("pH"), Browsable(True)>
    Property PH As Double

    <DisplayName("Conductividad"), Browsable(True)>
    Property Conductivity As Double

    <DisplayName("Residuos Secos"), Browsable(True)>
    Property DryWaste As Double

    <DisplayName("Alcalinidad de Bicarbonato"), Browsable(True)>
    Property BicarbonateAlkalinity As Double

    <DisplayName("Alcalinidad de Carbonato"), Browsable(True)>
    Property CarbonateAlkalinity As Double

    <DisplayName("Cloruros"), Browsable(True)>
    Property Chlorides As Double

    <DisplayName("Nitratos"), Browsable(True)>
    Property Nitrates As Double

    <DisplayName("Sulfatos"), Browsable(True)>
    Property Sulfates As Double

    <DisplayName("Calcio"), Browsable(True)>
    Property Calcium As Double

    <DisplayName("Magnesio"), Browsable(True)>
    Property Magnesium As Double

    <DisplayName("Sulfuros Totales (HS-)"), Browsable(True)>
    Property TotalSulfur As Double

    <DisplayName("Potasio"), Browsable(True)>
    Property Potassium As Double

    <DisplayName("Sodio"), Browsable(True)>
    Property Sodium As Double

    <DisplayName("Fluoruros"), Browsable(True)>
    Property Fluorides As Double

    <DisplayName("DRO"), Browsable(True)>
    Property DRO As Double

    <DisplayName("GRO"), Browsable(True)>
    Property GRO As Double

    <DisplayName("MRO"), Browsable(True)>
    Property MRO As Double

    <DisplayName("Hidrocarburos totales (EPA 8015)"), Browsable(True)>
    Property TotalHydrocarbons_EPA8015 As Double

    <DisplayName("Hidrocarburos totales (TNRCC 1005)"), Browsable(True)>
    Property TotalHydrocarbons_TNRCC1005 As Double

    <DisplayName("Benceno"), Browsable(True)>
    Property Benzene As Double

    <DisplayName("Tolueno"), Browsable(True)>
    Property Tolueno As Double

    <DisplayName("Etilbenceno"), Browsable(True)>
    Property Ethylbenzene As Double

    <DisplayName("Xileno (o)"), Browsable(True)>
    Property XyleneO As Double

    <DisplayName("Xileno (p-m)"), Browsable(True)>
    Property XylenePM As Double

    <DisplayName("Xileno total"), Browsable(True)>
    Property TotalXylene As Double

    <DisplayName("Naftaleno"), Browsable(True)>
    Property Naphthalene As Double

    <DisplayName("Acenafteno"), Browsable(True)>
    Property Acenafthene As Double

    <DisplayName("Acenaftileno"), Browsable(True)>
    Property Acenaphthylene As Double

    <DisplayName("Fluoreno"), Browsable(True)>
    Property Fluorene As Double

    <DisplayName("Antraceno"), Browsable(True)>
    Property Anthracene As Double

    <DisplayName("Fenantreno"), Browsable(True)>
    Property Fenanthrene As Double

    <DisplayName("Fluoranteno"), Browsable(True)>
    Property Fluoranthene As Double

    <DisplayName("Pireno"), Browsable(True)>
    Property Pyrene As Double

    <DisplayName("Benzo(a)antraceno"), Browsable(True)>
    Property BenzoAAnthracene As Double

    <DisplayName("Criseno"), Browsable(True)>
    Property Crysene As Double

    <DisplayName("Benzo(a)pireno"), Browsable(True)>
    Property BenzoAPyrene As Double

    <DisplayName("Benzo(b)fluoranteno"), Browsable(True)>
    Property BenzoBFluoranthene As Double

    <DisplayName("Benzo(g,h,i)perileno"), Browsable(True)>
    Property BenzoGHIPerylene As Double

    <DisplayName("Benzo(k)fluoranteno"), Browsable(True)>
    Property BenzoKFluoranthene As Double

    <DisplayName("Dibenzo(a,h)antraceno"), Browsable(True)>
    Property DibenzoAHAnthracene As Double

    <DisplayName("Indeno(1,2,3-cd)pireno"), Browsable(True)>
    Property Indeno123CDPyrene As Double

    <DisplayName("Arsénico"), Browsable(True)>
    Property Arsenic As Double

    <DisplayName("Cadmio"), Browsable(True)>
    Property Cadmium As Double

    <DisplayName("Cobalto"), Browsable(True)>
    Property Cobalt As Double

    <DisplayName("Cobre"), Browsable(True)>
    Property Copper As Double

    <DisplayName("Cromo total"), Browsable(True)>
    Property TotalChrome As Double

    <DisplayName("Mercurio"), Browsable(True)>
    Property Mercury As Double

    <DisplayName("Níquel"), Browsable(True)>
    Property Nickel As Double

    <DisplayName("Plomo"), Browsable(True)>
    Property Lead As Double

    <DisplayName("Selenio"), Browsable(True)>
    Property Selenium As Double

    <DisplayName("Zinc"), Browsable(True)>
    Property Zinc As Double

    Public Overrides Function GetChemicalAnalysisType(propertyName As String) As ChemicalAnalysisType
        Return WaterAnalysisTypes(propertyName)
    End Function

    Shared ReadOnly Property Properties As New Dictionary(Of String, String) From {
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.PH)), NameOf(WaterAnalysis.PH)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Conductivity)), NameOf(WaterAnalysis.Conductivity)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.DryWaste)), NameOf(WaterAnalysis.DryWaste)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BicarbonateAlkalinity)), NameOf(WaterAnalysis.BicarbonateAlkalinity)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.CarbonateAlkalinity)), NameOf(WaterAnalysis.CarbonateAlkalinity)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Chlorides)), NameOf(WaterAnalysis.Chlorides)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Nitrates)), NameOf(WaterAnalysis.Nitrates)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Sulfates)), NameOf(WaterAnalysis.Sulfates)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Calcium)), NameOf(WaterAnalysis.Calcium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Magnesium)), NameOf(WaterAnalysis.Magnesium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.TotalSulfur)), NameOf(WaterAnalysis.TotalSulfur)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Potassium)), NameOf(WaterAnalysis.Potassium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Sodium)), NameOf(WaterAnalysis.Sodium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Fluorides)), NameOf(WaterAnalysis.Fluorides)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.DRO)), NameOf(WaterAnalysis.DRO)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.GRO)), NameOf(WaterAnalysis.GRO)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.MRO)), NameOf(WaterAnalysis.MRO)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.TotalHydrocarbons_EPA8015)), NameOf(WaterAnalysis.TotalHydrocarbons_EPA8015)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.TotalHydrocarbons_TNRCC1005)), NameOf(WaterAnalysis.TotalHydrocarbons_TNRCC1005)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Benzene)), NameOf(WaterAnalysis.Benzene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Tolueno)), NameOf(WaterAnalysis.Tolueno)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Ethylbenzene)), NameOf(WaterAnalysis.Ethylbenzene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.XyleneO)), NameOf(WaterAnalysis.XyleneO)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.XylenePM)), NameOf(WaterAnalysis.XylenePM)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.TotalXylene)), NameOf(WaterAnalysis.TotalXylene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Naphthalene)), NameOf(WaterAnalysis.Naphthalene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Acenafthene)), NameOf(WaterAnalysis.Acenafthene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Acenaphthylene)), NameOf(WaterAnalysis.Acenaphthylene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Fluorene)), NameOf(WaterAnalysis.Fluorene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Anthracene)), NameOf(WaterAnalysis.Anthracene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Fenanthrene)), NameOf(WaterAnalysis.Fenanthrene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Fluoranthene)), NameOf(WaterAnalysis.Fluoranthene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Pyrene)), NameOf(WaterAnalysis.Pyrene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BenzoAAnthracene)), NameOf(WaterAnalysis.BenzoAAnthracene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Crysene)), NameOf(WaterAnalysis.Crysene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BenzoAPyrene)), NameOf(WaterAnalysis.BenzoAPyrene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BenzoBFluoranthene)), NameOf(WaterAnalysis.BenzoBFluoranthene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BenzoGHIPerylene)), NameOf(WaterAnalysis.BenzoGHIPerylene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.BenzoKFluoranthene)), NameOf(WaterAnalysis.BenzoKFluoranthene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.DibenzoAHAnthracene)), NameOf(WaterAnalysis.DibenzoAHAnthracene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Indeno123CDPyrene)), NameOf(WaterAnalysis.Indeno123CDPyrene)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Arsenic)), NameOf(WaterAnalysis.Arsenic)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Cadmium)), NameOf(WaterAnalysis.Cadmium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Copper)), NameOf(WaterAnalysis.Copper)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Cobalt)), NameOf(WaterAnalysis.Cobalt)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.TotalChrome)), NameOf(WaterAnalysis.TotalChrome)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Mercury)), NameOf(WaterAnalysis.Mercury)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Nickel)), NameOf(WaterAnalysis.Nickel)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Lead)), NameOf(WaterAnalysis.Lead)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Selenium)), NameOf(WaterAnalysis.Selenium)},
        {GetDisplayName(Of WaterAnalysis)(NameOf(WaterAnalysis.Zinc)), NameOf(WaterAnalysis.Zinc)}}
End Class

File: Model/YPFModel/src/WaterAnalysis.cs
using System.ComponentModel;
using Wells.BaseModel.Models;

namespace Wells.YPFModel
{
    public class WaterAnalysis : ChemicalAnalysis
    {
        public WaterAnalysis() : base() { SampleOf = SampleType.Water; }

        public WaterAnalysis(Well well) : base(well) { SampleOf = SampleType.Water; }

        public override ChemicalAnalysisType GetChemicalAnalysisType(string propertyName)
        {
            return WaterAnalysisTypes[propertyName];
        }

        #region Properties
        [DisplayName("pH"), Browsable(true)]
        public double PH { get; set; }

        [DisplayName("Conductividad"), Browsable(true)]
        public double Conductivity { get; set; }

        [DisplayName("Residuos Secos"), Browsable(true)]
        public double DryWaste { get; set; }

        [DisplayName("Alcalinidad de Bicarbonato"), Browsable(true)]
        public double BicarbonateAlkalinity { get; set; }

        [DisplayName("Alcalinidad de Carbonato"), Browsable(true)]
        public double CarbonateAlkalinity { get; set; }

        [DisplayName("Cloruros"), Browsable(true)]
        public double Chlorides { get; set; }

        [DisplayName("Nitratos"), Browsable(true)]
        public double Nitrates { get; set; }

        [DisplayName("Sulfatos"), Browsable(true)]
        public double Sulfates { get; set; }

        [DisplayName("Calcio"), Browsable(true)]
        public double Calcium { get; set; }

        [DisplayName("Magnesio"), Browsable(true)]
        public double Magnesium { get; set; }

        [DisplayName("Sulfuros Totales (HS-)"), Browsable(true)]
        public double TotalSulfur { get; set; }

        [DisplayName("Potasio"), Browsable(true)]
        public double Potassium { get; set; }

        [DisplayName("Sodio"), Browsable(true)]
        public double Sodium { get; set; }

        [DisplayName("Fluoruros"), Browsable(true)]
        public double Fluorides { get; set; }

        [DisplayName("DRO"), Browsable(true)]
        public double DRO { get; set; }

        [DisplayName("GRO"), Browsable(true)]
        public double GRO { get; set; }

        [DisplayName("MRO"), Browsable(true)]
        public double MRO { get; set; }

        [DisplayName("Hidrocarburos totales (EPA 8015)"), Browsable(true)]
        public double TotalHydrocarbons_EPA8015 { get; set; }

        [DisplayName("Hidrocarburos totales (TNRCC 1005)"), Browsable(true)]
        public double TotalHydrocarbons_TNRCC1005 { get; set; }

        [DisplayName("Benceno"), Browsable(true)]
        public double Benzene { get; set; }

        [DisplayName("Tolueno"), Browsable(true)]
        public double Tolueno { get; set; }

        [DisplayName("Etilbenceno"), Browsable(true)]
        public double Ethylbenzene { get; set; }

        [DisplayName("Xileno (o)"), Browsable(true)]
        public double XyleneO { get; set; }

        [DisplayName("Xileno (p-m)"), Browsable(true)]
        public double XylenePM { get; set; }

        [DisplayName("Xileno total"), Browsable(true)]
        public double TotalXylene { get; set; }

        [DisplayName("Naftaleno"), Browsable(true)]
        public double Naphthalene { get; set; }

        [DisplayName("Acenafteno"), Browsable(true)]
        public double Acenafthene { get; set; }

        [DisplayName("Acenaftileno"), Browsable(true)]
        public double Acenaphthylene { get; set; }

        [DisplayName("Fluoreno"), Browsable(true)]
        public double Fluorene { get; set; }

        [DisplayName("Antraceno"), Browsable(true)]
        public double Anthracene { get; set; }

        [DisplayName("Fenantreno"), Browsable(true)]
        public double Fenanthrene { get; set; }

        [DisplayName("Fluoranteno"), Browsable(true)]
        public double Fluoranthene { get; set; }

        [DisplayName("Pireno"), Browsable(true)]
        public double Pyrene { get; set; }

        [DisplayName("Benzo(a)antraceno"), Browsable(true)]
        public double BenzoAAnthracene { get; set; }

        [DisplayName("Criseno"), Browsable(true)]
        public double Crysene { get; set; }

        [DisplayName("Benzo(a)pireno"), Browsable(true)]
        public double BenzoAPyrene { get; set; }

        [DisplayName("Benzo(b)fluoranteno"), Browsable(true)]
        public double BenzoBFluoranthene { get; set; }

        [DisplayName("Benzo(g,h,i)perileno"), Browsable(true)]
        public double BenzoGHIPerylene { get; set; }

        [DisplayName("Benzo(k)fluoranteno"), Browsable(true)]
        public double BenzoKFluoranthene { get; set; }

        [DisplayName("Dibenzo(a,h)antraceno"), Browsable(true)]
        public double DibenzoAHAnthracene { get; set; }

        [DisplayName("Indeno(1,2,3-cd)pireno"), Browsable(true)]
        public double Indeno123CDPyrene { get; set; }

        [DisplayName("Arsénico"), Browsable(true)]
        public double Arsenic { get; set; }

        [DisplayName("Cadmio"), Browsable(true)]
        public double Cadmium { get; set; }

        [DisplayName("Cobalto"), Browsable(true)]
        public double Cobalt { get; set; }

        [DisplayName("Cobre"), Browsable(true)]
        public double Copper { get; set; }

        [DisplayName("Cromo total"), Browsable(true)]
        public double TotalChrome { get; set; }

        [DisplayName("Mercurio"), Browsable(true)]
        public double Mercury { get; set; }

        [DisplayName("Níquel"), Browsable(true)]
        public double Nickel { get; set; }

        [DisplayName("Plomo"), Browsable(true)]
        public double Lead { get; set; }

        [DisplayName("Selenio"), Browsable(true)]
        public double Selenium { get; set; }

        [DisplayName("Zinc"), Browsable(true)]
        public double Zinc { get; set; }
        #endregion
    }
}
